common/PerfStressShared/PerfStressUtilities.cs

```diff
@@ -0,0 +1,133 @@
+            var mb = ab.DefineDynamicModule("Options");
+
+            foreach (var t in testTypes)
+            {
+                var baseOptionsType = t.GetConstructors().First().GetParameters()[0].ParameterType;
+                var tb = mb.DefineType(t.Name + "Options", TypeAttributes.Public, baseOptionsType);
+
+                var attrCtor = typeof(VerbAttribute).GetConstructor(new Type[] { typeof(string), typeof(bool) });
+                var verbName = GetVerbName(t.Name);
+                tb.SetCustomAttribute(new CustomAttributeBuilder(attrCtor,
+                    new object[] { verbName, attrCtor.GetParameters()[1].DefaultValue }));
+
+                optionTypes.Add(tb.CreateType());
+            }
+
+            return optionTypes.ToArray();
+        }
+
+        internal static string GetVerbName(string testName)
+        {
+            // Remove suffix "Test" if exists
+            return Regex.Replace(testName, "Test$", string.Empty, RegexOptions.IgnoreCase);
+        }
+
+        // Run in dedicated thread instead of using async/await in ThreadPool, to ensure this thread has priority
+        // and never fails to run to due ThreadPool starvation.
+        internal static Thread PrintStatus(string header, Func<object> status, bool newLine, CancellationToken token, int intervalSeconds = 1)
+        {
+            var thread = new Thread(() =>
+            {
+                Console.WriteLine(header);
+
+                bool needsExtraNewline = false;
+
+                while (!token.IsCancellationRequested)
+                {
+                    try
+                    {
+                        Sleep(TimeSpan.FromSeconds(intervalSeconds), token);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                    }
+
+                    var obj = status();
+
+                    if (newLine)
+                    {
+                        Console.WriteLine(obj);
+                    }
+                    else
+                    {
+                        Console.Write(obj);
+                        needsExtraNewline = true;
+                    }
+                }
+
+                if (needsExtraNewline)
+                {
+                    Console.WriteLine();
+                }
+
+                Console.WriteLine();
+            });
+
+            thread.Start();
+
+            return thread;
+        }
+
+        internal static void Sleep(TimeSpan timeout, CancellationToken token)
+        {
+            var sw = Stopwatch.StartNew();
+            while (sw.Elapsed < timeout)
+            {
+                if (token.IsCancellationRequested)
+                {
+                    // Simulate behavior of Task.Delay(TimeSpan, CancellationToken)
+                    throw new OperationCanceledException();
+                }
+
+                Thread.Sleep(TimeSpan.FromMilliseconds(10));
```

Comment:
This causes lots of context switching and can hinder perf. Instead, use:

```c#
if (0 == WaitHandle.WaitAny(token.WaitHandle, timeout))
{
  throw new OperationCanceledException();
}
```

Reply:
#16972

Reply:
Review comments like this one were only a few-line changes. Any reason to not just push a change now as opposed to a bug that could go stale? Especially in a perf framework, sleeping like this is impactful.

Reply:
I will fix this immediately after this PR is merged.  I prefer to keep my PRs scoped to a single logical change, and address unrelated (but much appreciated) feedback in separate PRs.  I agree this might be impactful and I want to test the perf impact as well.